### PR TITLE
Make helper functions in CoreDataModelable public

### PR DESCRIPTION
**Pull request by rcedwards**
This PR:
* renames `entityInContext` to `entityDescriptionInContext`
* makes `entityDescriptionInContext` a public function
* makes `fetchRequestForEntity` a public function
* enables code coverage gathering for all targets

Addresses #89

### CODE REVIEW

**Comment by wolfbrain** [comment]
:eyes: @jeremy-w

File: CoreDataStack/CoreDataModelable.swift [hunk 1]
```diff
@@ -37,10 +37,19 @@ extension CoreDataModelable where Self: NSManagedObject {
...
 
-    static func entityInContext(context: NSManagedObjectContext) -> NSEntityDescription! {
+    // MARK: - Finding Objects
+
+    /**
+    Creates an `NSEntityDescription` of the `CoreDataModelable` entity using the `entityName`
+
+    - parameter context: `NSManagedObjectContext` to create the object within.
+
+    - returns: `NSEntityDescription`: The entity description.
```
Comment (jeremy-w) [inline CoreDataStack/CoreDataModelable.swift]: The interesting thing here is the ImplicitlyUnwrappedOptional behavior. That it returns an entity description we can read from the type signature.

File: CoreDataStack/CoreDataModelable.swift [hunk 2]
```diff
@@ -37,10 +37,19 @@ extension CoreDataModelable where Self: NSManagedObject {
...
     public init(managedObjectContext context: NSManagedObjectContext) {
-        self.init(entity: Self.entityInContext(context), insertIntoManagedObjectContext: context)
+        self.init(entity: Self.entityDescriptionInContext(context), insertIntoManagedObjectContext: context)
     }
 
-    static func entityInContext(context: NSManagedObjectContext) -> NSEntityDescription! {
+    // MARK: - Finding Objects
+
+    /**
+    Creates an `NSEntityDescription` of the `CoreDataModelable` entity using the `entityName`
```
Comment (jeremy-w) [inline CoreDataStack/CoreDataModelable.swift]: The reader's first question will be, "Why not use `NSEntityDescription.entityForName(_:inManagedObjectContext:)`?" Answering that question would be a significant help.

File: CoreDataStack/CoreDataModelable.swift [hunk 3]
```diff
@@ -37,10 +37,19 @@ extension CoreDataModelable where Self: NSManagedObject {
...
+
+    - parameter context: `NSManagedObjectContext` to create the object within.
+
+    - returns: `NSEntityDescription`: The entity description.
+    */
+    static public func entityDescriptionInContext(context: NSManagedObjectContext) -> NSEntityDescription! {
```
Comment (jeremy-w) [inline CoreDataStack/CoreDataModelable.swift]: Going off the docstring, should this be `createEntityDescriptionIn(_:)` instead? (I'm also dropping the `Context` bit as a noise-word relative to the known parameter type.)

**Pull request merged**